File: src/contests/cg/puzzle/perl/horse-racing-duals.pl
# 🎮 CodinGame Puzzle - horse-racing-duals
# https://www.codingame.com/training/easy/horse-racing-duals

my $n = <STDIN>;
chomp $n;
my @s;
for (1 .. $n) {
    my $v = <STDIN>;
    chomp $v;
    push @s, $v;
}
@s = sort { $a <=> $b } @s;
my $min_diff = 9**9**9;
for my $i (1 .. $#s) {
    my $diff = $s[$i] - $s[$i - 1];
    $min_diff = $diff if $diff < $min_diff;
}
print "$min_diff\n";

File: src/contests/cg/puzzle/php/horse-racing-duals.php
<?php
// 🎮 CodinGame Puzzle - horse-racing-duals
// https://www.codingame.com/training/easy/horse-racing-duals

$n = (int) trim(fgets(STDIN));
$s = [];
for ($i = 0; $i < $n; $i++) {
    $s[] = (int) trim(fgets(STDIN));
}
sort($s);
$minDiff = PHP_INT_MAX;
for ($i = 1; $i < $n; $i++) {
    $diff = $s[$i] - $s[$i - 1];
    if ($diff < $minDiff) {
        $minDiff = $diff;
    }
}
echo $minDiff . "\n";
